components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -718,10 +718,10 @@ private void setBlobValue(String value, PreparedStatement prepStmt, int index) t
             if (inputStream != null) {
                 prepStmt.setBinaryStream(index, inputStream, inputStream.available());
             } else {
-                prepStmt.setBinaryStream(index, inputStream, 0);
+                prepStmt.setBinaryStream(index, new ByteArrayInputStream(CharacterEncoder.getSafeText("").getBytes()), 0);
             }
         } else {
-            prepStmt.setBinaryStream(index, null, 0);
+            prepStmt.setBinaryStream(index, new ByteArrayInputStream(CharacterEncoder.getSafeText("").getBytes()), 0);
```

Comment:
Adding the charset name when getting bytes would have been better to make this platfrom independent. Also do we really need to getSafeText on empty String?